e2etests/e2etests.js

```diff
@@ -41,12 +41,16 @@ device_provision(hubConnectionString, function (err, provisionedDevices) {
     console.log('Unable to create the devices needed.');
   } else {
     provisionedDevices.forEach(function(deviceToTest) {
-      acknowledgementProtocols.forEach(function (protocolToTest) {
-        device_acknowledge_tests(hubConnectionString, protocolToTest, deviceToTest);
+      if (deviceToTest.authenticationDescription !== 'shared private key') {
```

Comment:
I like performance improvements as much as the next guy but i don't like bad/hard to maintain code and one doesn't have to come at the expense of the other, which is what is happening here. You may check this in as is but it doesn't change the fact that an overhaul of this part of the code is needed. I guess you just don't want to take care of it.